extension/hook.js

```diff
@@ -1,30 +1,58 @@
 const getManifest = chrome.runtime.getManifest;
 const version = (getManifest && getManifest().version) || 'electron-version';
+let passedApolloConnected = false;
 
 const js = `
+let isConnected = false;
+
+const hookLogger = (stateObj) => {
+  if (!!window.__APOLLO_CLIENT__) {
+    const trimmedObj = {
+      queries: stateObj.state.queries,
+      mutations: stateObj.state.mutations
+    }
+    window.postMessage({ trimmedObj }, '*');
+  }
+}
+
 window.__APOLLO_DEVTOOLS_GLOBAL_HOOK__ = { version: "${version}" };
 
 let __APOLLO_POLL_COUNT__ = 0;
 const __APOLLO_POLL__ = setInterval(() => {
   if (!!window.__APOLLO_CLIENT__) {
-    window.postMessage({ APOLLO_CONNECTED: true }, '*');
+    window.postMessage({ APOLLO_CONNECTED: true}, '*');
+    isConnected = true;
+    window.__APOLLO_CLIENT__.__actionHookForDevTools(hookLogger)
     clearInterval(__APOLLO_POLL__);
   } else {
     __APOLLO_POLL_COUNT__ += 1;
   }
-
   if (__APOLLO_POLL_COUNT__ > 20) clearInterval(__APOLLO_POLL__);
 }, 500);
 `;
 
-var script = document.createElement('script');
+let script = document.createElement('script');
 script.textContent = js;
 document.documentElement.appendChild(script);
 script.parentNode.removeChild(script);
 
+// event.data has the data being passed in the message
 window.addEventListener('message', event => {
-  if (event.source != window) return;
-  if (!event.data.APOLLO_CONNECTED) return;
+  if (event.source != window) 
+    return;
 
-  chrome.runtime.sendMessage({ APOLLO_CONNECTED: true });
+  if (event.data.APOLLO_CONNECTED) {
+    if (!passedApolloConnected) {
+      chrome.runtime.sendMessage({ APOLLO_CONNECTED: true}, function() {
+        passedApolloConnected = true;
+      });
+    }
+  }
+
+  if (!!event.data.trimmedObj) {
+    chrome.runtime.sendMessage({ trimmedObj: event.data.trimmedObj });
+  }
+  else {
```

Comment:
I don't think you need this `else` statement - a blank return should have the same effect as no return